client/dimensions_api.go

```diff
@@ -16,10 +16,16 @@ import (
 const packageName = "client.DatasetAPI"
 
 // ErrHostEmpty is returned when a Client is created with an empty Host
-var ErrHostEmpty = errors.New("validation error: api host is required but was empty")
+var ErrHostEmpty = errors.New("api host is required but was empty")
+
+// ErrNoDatasetAPI is returned when a dataset API Client is not provided
+var ErrNoDatasetAPI = errors.New("dataset api is required but is not provided")
+
+// ErrNoDatastore is returned when a data store is not provided
+var ErrNoDatastore = errors.New("data store is required but is not provided")
 
 // ErrInstanceIDEmpty is returned when an InstanceID is not provided to a call that requires it
-var ErrInstanceIDEmpty = errors.New("validation error: instance id is required but is empty")
+var ErrInstanceIDEmpty = errors.New("instance id is required but is empty")
```

Comment:
This gets used in 4 different places which means that with the way that errors are reported in the logs it will be almost impossible to determine which one of the 4 possible places produced this error.
I ran into this problem within the last month of duplicate error messages and could not from examining the code determine where the error came from.
I think the above line needs to be changed from:
var ErrInstanceIDEmpty = errors.New("instance id is required but is empty")
to:
var ErrInstanceIDEmpty1 = errors.New("instance id is required but is empty-1")
var ErrInstanceIDEmpty2 = errors.New("instance id is required but is empty-2")
var ErrInstanceIDEmpty3 = errors.New("instance id is required but is empty-3")
var ErrInstanceIDEmpty4 = errors.New("instance id is required but is empty-4")
and any test code changed accordingly.
-=-=-
A similar change is needed for line 31 for: ErrDimensionNil
which is used twice.
-=-=-
Also, if there are any other places that use an error string more than once ... then that needs adjusting.

Reply:
thanks! maybe I should just remove the constants, or make sure to wrap the errors with a clearer text (not just 'validation error')

Reply:
Yes, i think removing the constants and using different strings in the code would be better as that way others in the future 'should' be less likely to re-use them and create more examples of the issue that needs fixing here.
Thanks.